lib/stores/S3Store.js

```diff
@@ -373,31 +397,44 @@ class S3Store extends DataStore {
             .promise()
             .then((data) => {
                 if (data.NextPartNumberMarker) {
-                    return this._retrieveParts(file_id, data.NextPartNumberMarker)
-                        .then((val) => [].concat(data.Parts, val));
+                    return this._retrieveParts(file_id, data.NextPartNumberMarker).then((val) =>
+                        [].concat(data.Parts, val)
+                    );
                 }
-
                 return data.Parts;
+            })
+            .then((parts) => {
+                // sort and filter only for call where `part_number_marker` is not set
+                if (part_number_marker === undefined) {
+                    parts.sort((a, b) => {
+                        return a.PartNumber - b.PartNumber;
+                    });
+
+                    parts = parts.filter((value, index) => {
```

Comment:
Why is this filtering necessary? It ensures that we have a consecutive number of parts, but is this important? Can't I have part numbers 1, 2, 3, 5, 6?

Reply:
Good question, this is from @mitjap's implementation and I'm not sure why we need it tbh

Reply:
Ok, I don't think it is necessary but let's leave it for now